esti/presign_test.go

```diff
@@ -1,20 +1,43 @@
 package esti
 
 import (
+	"io"
+	"net/http"
 	"net/url"
 	"strings"
 	"testing"
 
+	"github.com/treeverse/lakefs/pkg/block"
+
+	"github.com/thanhpk/randstr"
+
 	"github.com/go-openapi/swag"
-	"github.com/spf13/viper"
 	"github.com/stretchr/testify/require"
 	"github.com/treeverse/lakefs/pkg/api"
-	"github.com/treeverse/lakefs/pkg/block"
 )
 
+func matchPreSignedURLContent(t *testing.T, preSignedURL, content string) {
+	// Running an HTTP GET on the URL should return the object's content in body.
+	r, err := http.Get(preSignedURL)
+	require.NoErrorf(t, err, "failed to GET pre-signed url - %s", preSignedURL)
+
+	retrievedData, err := io.ReadAll(r.Body)
+	require.NoErrorf(t, err, "failed to read GET body from pre-signed url - %s", preSignedURL)
+	require.Equal(t, string(retrievedData), content, "pre-signed url body doesn't match uploaded content")
+	require.NoError(t, r.Body.Close(), "could not close response body")
+}
+
 func TestPreSign(t *testing.T) {
```

Comment:
Non-blocking: consider also a test of uploading data to a url retrieved by `GetPhysicalAddress`. I think that's the last missing piece.